datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -306,37 +307,99 @@ func (c *Consumer) updateOffset(buffer [][]*Segment, lastMsg *sarama.ConsumerMes
 	return nil
 }
 
+// renewExpireMsg updates expiration timer and message if the expiring message is handled.
+func (c *Consumer) renewExpireMsg(buffer [][]*Segment, timer *time.Timer, oldestMsg *sarama.ConsumerMessage) *sarama.ConsumerMessage {
+	if c.config.ExpirationTime > time.Duration(0) {
+		if len(buffer) == 0 {
+			timer.Stop()
+			return nil
+		} else if oldestMsg != buffer[0][0].orig {
+			timer.Reset(c.config.ExpirationTime)
+			return buffer[0][0].orig
+		} else {
+			return oldestMsg
+		}
+	}
+	return nil
+}
```

Comment:
I'm sorry I couldn't give you a good review.
How do you think of the style below?
I learned that more else makes it less legible in Clean code.
```
func (c *Consumer) renewExpireMsg(buffer [][]*Segment, timer *time.Timer, oldestMsg *sarama.ConsumerMessage) *sarama.ConsumerMessage {
	if c.config.ExpirationTime <= time.Duration(0) {
		return nil
	}
	if len(buffer) == 0 {
		timer.Stop()
		return nil
	}
	if oldestMsg != buffer[0][0].orig {
		timer.Reset(c.config.ExpirationTime)
		return buffer[0][0].orig
	}
	return oldestMsg
}
```

Reply:
Applied. Thanks :)